x-pack/platform/plugins/shared/lens/public/app_plugin/shared/edit_on_the_fly/lens_configuration_flyout.tsx

```diff
@@ -97,7 +97,8 @@ export function LensEditConfigurationFlyout({
             previousAttrs.state.datasourceStates[datasourceId],
             previousAttrs.references,
             datasourceStates[datasourceId].state,
-            attributes.references
+            datasourceMap[datasourceId].getPersistableState(datasourceStates[datasourceId].state)
```

Comment:
Seems to work fine, do you mind adding a comment to specify why we need to get the references from the persisted state, rather then the attributes.references?